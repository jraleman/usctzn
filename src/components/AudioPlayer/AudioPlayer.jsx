import React from 'react';
import styled from 'styled-components';
import DownloadLink from '../DownloadLink';
import uniqueId from '../../utils/helpers/uniqueId';

// TODO: Use an actual audio player instead of download mp3 file
const AudioPlayer = ({ audioFile }) => {
    const { src, label } = audioFile || {};
    return (
        <Container>
            <DownloadLink key={uniqueId('mp3_')} src={src} label={`ðŸ”Š ${label} ðŸ”Š`} />
        </Container>
    );
};

const Container = styled.div``;

export default AudioPlayer;
