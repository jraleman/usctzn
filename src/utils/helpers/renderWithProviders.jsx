import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { lightTheme } from '../styles/theme';

const renderWithProviders = (Component) => {
    render(
        <ThemeProvider theme={lightTheme}>
            {Component}
        </ThemeProvider>
    )
};

export default renderWithProviders;
