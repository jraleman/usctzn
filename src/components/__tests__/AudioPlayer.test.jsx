import React from 'react';
import renderWithProviders from '../../utils/helpers/renderWithProviders';
import AudioPlayer from '../AudioPlayer';

describe('AudioPlayer', () => {
    it('renders', () => {
        const component = renderWithProviders(<AudioPlayer />);
        expect(component).not.toBeNull();
        expect(component).toMatchSnapshot();
    });
});
