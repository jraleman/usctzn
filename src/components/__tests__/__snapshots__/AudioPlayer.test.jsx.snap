// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudioPlayer renders 1`] = `undefined`;
